### PR TITLE
Fixes #212 - Fix: Validation Input & Results

### LINKED ISSUE #212: Secuity issue with search functionality
**Received this from our internal security team after talking DNA with them:** 
 
As discussed, please see the PoC for sanitizing the user input. The below code uses regex to remove any characters that are not letters or numbers. I know this is not considered to be the nicest way of handling strings, but it showcases the idea well enough.
 
The reason for using the whitelisting solution is that any blacklisting solution might leave out some special characters. Even though the UrlEncode-like functions are widely used, in this case I didn’t check the underlying library to be 100% sure, it handles any user input properly.
 
The usual attack vectors to be tried against a library are 1. Special characters (XSS, SQL injection, Object injection, etc.) and 2. Too long strings (eg. Buffer overflows). As the below code allows maximum 20 characters long alphanumeric string, it should be safe to use. It can of course be longer as well, 20 is just an arbitrary number here, as I didn’t see anyone searching for 20+ characters with non-malicious intent realistically.
 
The client-side sanitization works in this case, as the client has to explicitly turn off part of the code, in order to bypass this. While this could be done by an attacker, this could only result in a self-hack (attacker can’t turn off victim’s code partially).
 
I hope I wrote the code bugless and everything is clear, but should any of this not be true, feel free to contact me anytime.
 
Many thanks and best regards,
Norbert
 
https://github.com/youseedk/dna/blob/master/fractal-theme/assets/js/src/search-results.js line #32:
const searchString = getSearchParams.get('q');
 
-->
 ```
function allow_chars_and_nums_only(var str) {
                str = str.replace(/[^\w\s]/gi, '');
                return str.slice(0,20);
}
  
const searchString = allow_chars_and_nums_only(getSearchParams.get('q'));
 ```

### PR DESCRIPTION
**Pull request by Dechowmedia**
Fixes and hardens the search input as such we are limiting the characters (no special chars): 
- We are hardening the search field from [XSS injection](https://www.owasp.org/index.php/Cross-site_Scripting_(XSS)) (both on Results page + Sidebar)
- The search results page (with the `?q=` parameter) will be filtered through a regex - before being passed directly to the input thereby removing the potential of [Input Escaping](https://security.openstack.org/guidelines/dg_cross-site-scripting-xss.html)

### CODE REVIEW

File: fractal-theme/assets/js/src/search-results.js [hunk 1]
```diff
@@ -1,35 +1,43 @@
-import Fuse from 'fuse.js'
+import Fuse from "fuse.js";
```
Comment (havgry) [inline fractal-theme/assets/js/src/search-results.js]: Did you _prettier_ all over this file? :)

Comment (Dechowmedia) [reply fractal-theme/assets/js/src/search-results.js]: Its pretty right?

Comment (Dechowmedia) [reply fractal-theme/assets/js/src/search-results.js]: Adding a prettier file cause I wont cause a mess :'D

**Comment by grandorf** [comment]
> I don't see any problems with this. It's up to Michael to decide if he's ok with the prettier config as part of this PR :)

I'm fine with it.

**Comment by grandorf** [comment]
@Dechowmedia you should target the `develop` branch and not master. We use gitflow ;-)
https://dna.yousee.dk/docs/contributors

**Pull request merged**